src/Library.tsx

```diff
@@ -478,20 +479,24 @@ export default function Library() {
     });
   }
 
-  const chapterlistChapters = [];
+  useEffect(() => {
```

Comment:
This change isn't needed, right? I think only the change to the reducer is needed. `chapterlistChapters` doesn't need to be a state variable since it can be computed.

Reply:
By making this computed aren't you forcing a full comparison on said array for all changes to the reducer, while this will only force a reevaluation on a chapter change? That was my understanding, but I could be very wrong here.

Reply:
That is fair